### PR TITLE
Fix youtube and vimeo in the media gallery

### LINKED ISSUE #2791: Youtube in media gallery not working
## Type
- Non critical bug

## Problem description
When adding a Youtube video for the media gallery, it's not working in the frontend.

## Steps to reproduce
Add a media gallery in the backend, add an online movie to it and add Youtube video.
Add the widget from the media gallery to a page.
The YT video won't be shown: 
Screen Shot 2019-04-16 at 07 22 54

## Expected behavior / Proposed solutions
Youtube thumbnail shown with link to YT video (in modal or something).

### PR DESCRIPTION
**Pull request by carakas**
## Type

- Non critical bugfix

## Resolves the following issues

fixes #2791 

## Pull request description

The media gallery widgets didn't work with youtube and vimeo video's. That has been fixed now.
Also the video's now have thumbnails in the backend.

### CODE REVIEW

File: src/Backend/Modules/MediaLibrary/Component/StorageProvider/YoutubeStorageProvider.php [hunk 1]
```diff
@@ -8,7 +8,7 @@ class YoutubeStorageProvider extends MovieStorageProvider
 {
     public function getIncludeHTML(MediaItem $mediaItem): string
     {
-        return '<iframe width="560" height="315" src="' . $this->includeUrl . $mediaItem->getUrl() . '" frameborder="0" allowfullscreen></iframe>';
+        return '<iframe width="100%" height="100%" src="' . $this->includeUrl . $mediaItem->getUrl() . '" frameborder="0" allowfullscreen></iframe>';
```
Comment (StijnVrolijk) [inline src/Backend/Modules/MediaLibrary/Component/StorageProvider/YoutubeStorageProvider.php]: Does `height="100%"` even work? I'd run this by a frontender

Comment (carakas) [reply src/Backend/Modules/MediaLibrary/Component/StorageProvider/YoutubeStorageProvider.php]: it worked when I tried it here. It was full in the gallery and a small player when displayed as a single item.
If there is a better way a frontender may always suggest it
@forkcms/frontenders

**Comment by jeroendesloovere** [comment]
The only thing of feedback I can provide is that it should be better if we move all HTML outside of the PHP classes. Perhaps using a translation with %1$s to replace filename, etc...

Example, replacing
```php
return '<iframe src="' . $this->includeUrl . $mediaItem->getUrl() . '?color=ffffff&title=0&byline=0&portrait=0&badge=0" width="100%" height="100%" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>';
``` 

by
```php
return sprintf(
    $this->htmlForInclusion,
    $this->includeUrl . $mediaItem->getUrl()
);
```

**Comment by carakas** [comment]
@jeroendesloovere going to leave it like this. You can always change it in the php if needed. And with the width stuff it will take the size of it's parent so you can control it like that

**Pull request merged**